Patch 408 (#430)
aquasecurity/cloud-security-remediation-guides: en/aws/ec2/vpc-flow-logs-enabled.md

| **Recommended Action** | Enable VPC flow logs for each VPC |

## Detailed Remediation Steps
1. Log in to the AWS Management Console.
2. Select the "Services" option and search for VPC. </br> <img src="/resources/aws/ec2/vpc-flow-logs-enabled/step2.png"/>
3. Scroll down the left navigation panel and choose "Your VPCs" under "Virtual Private Cloud". </br> <img src="/resources/aws/ec2/vpc-flow-logs-enabled/step3.png"/>
4. Select the "VPC" that needs to be verified for "VPC Flow Logs". </br> <img src="/resources/aws/ec2/vpc-flow-logs-enabled/step4.png"/>
5. Scroll down the bottom dashboard panel and choose "Flow Logs" tab. If there are no flow logs then "No flow logs found in this Region" message will be displayed. </br> <img src="/resources/aws/ec2/vpc-flow-logs-enabled/step5.png"/>

> 6. Repeat steps number 2 - 6 to verify "Flow Logs" are enabled or not in other VPCs in the region. </br>

Reviewer: ```suggestion
6. Repeat steps number 2 - 6 to verify whether "Flow Logs" are enabled or not for other VPCs in the remaining regions. </br>
```

7. Navigate to "Your VPCs" in the left navigation panel, select the VPC and click on the "Flow Logs" tab in the bottom dashboard panel.</br> <img src="/resources/aws/ec2/vpc-flow-logs-enabled/step7.png"/>
8. Click on the "Create flow log" button to create the "VPC Flow Logs". </br> <img src="/resources/aws/ec2/vpc-flow-logs-enabled/step8.png"/>

> 9. In the "Create flow log" dialog box select the "Filter" that describes the type of traffic to be logged and select the " Maximum aggregation interval".</br> <img src="/resources/aws/ec2/vpc-flow-logs-enabled/step9.png"/>

Reviewer: ```suggestion
9. In the "Create flow log" dialog box select the "Filter" that describes the type of traffic to be logged and select the "Maximum aggregation interval".</br> <img src="/resources/aws/ec2/vpc-flow-logs-enabled/step9.png"/>
```

> 10. Select the destination to which the flow log data is to be published from the options.If the "S3 bucket" is selected for the log data to be published then provide the "S3 Bucket ARN" which is the ARN of the Amazon S3 bucket to which the flow log is to be published. Also select the "Log Record Format" & "Log File Format" along with the "Partition logs by time" and click on the "Create flow log" button at the bottom. </br> <img src="/resources/aws/ec2/vpc-flow-logs-enabled/step10.png"/>

Reviewer: ```suggestion
10. Select the destination to which the flow log data is to be published from the options. If the "S3 bucket" is selected then provide the "S3 Bucket ARN" in which the flow log is to be published. Also select the "Log Record Format" & "Log File Format" along with the "Partition logs by time" and click on the "Create flow log" button at the bottom. </br> <img src="/resources/aws/ec2/vpc-flow-logs-enabled/step10.png"/>
```
